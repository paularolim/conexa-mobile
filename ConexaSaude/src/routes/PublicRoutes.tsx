import { createNativeStackNavigator } from '@react-navigation/native-stack';

import { Login } from '@screens/Login';

import { PublicRoutesParamList } from './types';

const Stack = createNativeStackNavigator<PublicRoutesParamList>();

export function PublicRoutes() {
  return (
    <Stack.Navigator screenOptions={{ headerShown: false }}>
      <Stack.Screen name="Login" component={Login} />
    </Stack.Navigator>
  );
}
