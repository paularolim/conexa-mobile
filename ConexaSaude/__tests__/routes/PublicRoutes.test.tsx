import { NavigationContainer } from '@react-navigation/native';
import { cleanup } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import { PublicRoutes } from '@routes/PublicRoutes';

import { Wrapper } from '../Wrapper';

jest.useFakeTimers();

describe('PublicRoutes', () => {
  afterEach(() => cleanup());

  it('should render routes correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <NavigationContainer>
            <PublicRoutes />
          </NavigationContainer>
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
