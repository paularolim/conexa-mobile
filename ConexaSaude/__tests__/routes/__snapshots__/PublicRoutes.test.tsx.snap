// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicRoutes should render routes correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "gap": 20,
                  "justifyContent": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 24,
                  "paddingRight": 24,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "gap": 0,
                    "width": "100%",
                  },
                ]
              }
            >
              <Image
                source={
                  {
                    "testUri": "../../../src/assets/images/logo.png",
                  }
                }
                style={
                  [
                    {
                      "maxWidth": "60%",
                      "resizeMode": "contain",
                    },
                  ]
                }
              />
            </View>
            <View
              space={8}
              style={
                [
                  {
                    "gap": 8,
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                    },
                    {
                      "color": "#707683",
                      "fontSize": 20,
                      "fontWeight": "bold",
                    },
                  ]
                }
              >
                Para entrar, utilize as credenciais que vocÃª recebeu via e-mail
              </Text>
              <View
                style={
                  [
                    {
                      "flexDirection": "row",
                      "gap": 10,
                    },
                  ]
                }
              >
                <View
                  primary={true}
                  style={
                    [
                      {
                        "backgroundColor": "#F28080",
                        "borderRadius": 2,
                        "height": 5,
                        "width": 80,
                      },
                    ]
                  }
                />
                <View
                  style={
                    [
                      {
                        "backgroundColor": "#C2CFE0",
                        "borderRadius": 2,
                        "height": 5,
                        "width": 80,
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              space={16}
              style={
                [
                  {
                    "gap": 16,
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "flex-start",
                      "backgroundColor": "#FFFFFF",
                      "justifyContent": "center",
                      "width": "100%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                      },
                      {
                        "color": "#707683",
                        "fontSize": 18,
                      },
                    ]
                  }
                >
                  E-mail
                </Text>
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "borderColor": "#707683",
                        "borderRadius": 4,
                        "borderStyle": "solid",
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "minHeight": 48,
                        "paddingBottom": 8,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                        "paddingTop": 8,
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <TextInput
                    autoCapitalize="none"
                    keyboardType="email-address"
                    style={
                      [
                        {
                          "color": "#000000",
                          "flexGrow": 1,
                          "paddingBottom": 0,
                          "paddingLeft": 0,
                          "paddingRight": 0,
                          "paddingTop": 0,
                        },
                      ]
                    }
                    testID="field"
                  />
                </View>
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "flex-start",
                      "backgroundColor": "#FFFFFF",
                      "justifyContent": "center",
                      "width": "100%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                      },
                      {
                        "color": "#707683",
                        "fontSize": 18,
                      },
                    ]
                  }
                >
                  Senha
                </Text>
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "borderColor": "#707683",
                        "borderRadius": 4,
                        "borderStyle": "solid",
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "minHeight": 48,
                        "paddingBottom": 8,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                        "paddingTop": 8,
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <TextInput
                    autoCapitalize="none"
                    secureTextEntry={true}
                    style={
                      [
                        {
                          "color": "#000000",
                          "flexGrow": 1,
                          "paddingBottom": 0,
                          "paddingLeft": 0,
                          "paddingRight": 0,
                          "paddingTop": 0,
                        },
                      ]
                    }
                    testID="field"
                  />
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "opacity": 1,
                      }
                    }
                    testID="password-visibility-toggle"
                  >
                    <svg
                      data-file-name="SvgEyeOn"
                      height={25}
                      stroke="#707683"
                      width={25}
                    />
                  </View>
                </View>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#2ED47A",
                    "borderRadius": 4,
                    "justifyContent": "center",
                    "maxWidth": "100%",
                    "opacity": 1,
                    "paddingBottom": 8,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 8,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                      },
                      {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                      },
                    ]
                  }
                  type="filled"
                >
                  Entrar
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 4,
                    "justifyContent": "center",
                    "maxWidth": "100%",
                    "opacity": 1,
                    "paddingBottom": 8,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 8,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#000000",
                      },
                      {
                        "color": "#109CF1",
                        "fontSize": 18,
                      },
                    ]
                  }
                  type="ghost"
                >
                  Esqueci a senha
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Login"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
